examples/circuit_example.rs

```diff
@@ -0,0 +1,70 @@
+use ndarray::Array2;
+use quaru::{
+    math::real_arr_to_complex,
+    operation::{self, *},
+    register::*,
+};
+use stopwatch::Stopwatch;
+
+
+fn main() {
+
+
+    let mut reg = Register::new(&[false ;2]);
+
+    let mut circ = QuantumCircuit::new();
+    circ.add_operation(hadamard(0));
+    circ.add_operation(cnot(0, 1));
+
+    circ.reduce_gates_with_one_off_size(2);
```

Comment:
That is smart. However the type of optimization thet yeild the best result can differ depending oin the circuit

Reply:
Yeah that souds like a good idea. As of now its quite quick. But ive only tested smaller circuits. It migt be wirth making this an issue, exploring what kind of optimizations do what to different types of gates. 
Take shors as an example, all thosse gates are really hard to optimize, both because of size and how they are represented.